src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java

```diff
@@ -0,0 +1,152 @@
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(uri))
+                .setHeader("X-GEMINI-APIKEY",apiKey)
+                .setHeader("Content-Type","text/plain")
+                .setHeader("X-GEMINI-PAYLOAD",new String(b64,StandardCharsets.UTF_8))
+                .setHeader("X-GEMINI-SIGNATURE",signature)
+                .setHeader("Cache-Control","no-cache")
+                .method(reqType,HttpRequest.BodyPublishers.noBody())
+                .build();
+        HttpResponse<String> response = HttpClient.newHttpClient().send(request,HttpResponse.BodyHandlers.ofString());
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return new ResponseEntity<String>(response.body(),headers, HttpStatus.ACCEPTED);
+    }
+
+    @Override
+    public List<SymbolResDTO> getPairs() {
+        List<SymbolResDTO> symbolDetails = new ArrayList<>();
+        try {
+            ResponseEntity<String> res = apiCaller(baseUrl+"/v1/symbols","GET");
+            List<String> symbols= new ObjectMapper().readValue(res.getBody(), new TypeReference<List<String>>() {});
+            for(String symbol : symbols){
+                ResponseEntity<String> symRes = apiCaller(baseUrl+"/v1/symbols/details/"+symbol,"GET");
+                SymbolResponse response = new ObjectMapper().readValue(symRes.getBody(),SymbolResponse.class);
+                symbolDetails.add(GeminiUtil.getSymbolResDTO(response));
+            }
+            return symbolDetails;
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public ResponseEntity<String> accountInfo(String apiKey, String secretKey, String passphrase, String timestamp) {
+        return null;
+    }
+
+    @Override
+        public PlaceOrderResDTO createOrder(String apiKey, String secretKey, String passphrase, PlaceOrderReqDTO body) {
+        CreateOrderRequest request = GeminiUtil.getCreateOrderReqDTO(body);
+        request.setClientOrderId("12344545");
+        request.setNonce(getNonce());
+        request.setAccountType("primary");
+        request.setRequest("/v1/order/new");
+
+        try {
+            String json = new ObjectMapper().writeValueAsString(request);
+            byte[] b64 = getB64(json);
+            String signature = GeminiUtil.getSignature(b64,secretKey);
+            ResponseEntity<String> res = apiCaller(baseUrl+"/v1/order/new","POST",b64,signature,apiKey);
+
+            CreateOrderResponse orderResponse = new ObjectMapper().readValue(res.getBody(), CreateOrderResponse.class);
+            return GeminiUtil.getPlaceOrderResDTO(orderResponse);
+        }
+        catch (NoSuchAlgorithmException | InvalidKeyException | IOException | InterruptedException e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
Can we have some proper return value instead of null?

Reply:
If null then we can send appropriate response from controller